srecord/input/filter/message/adler32.cc

```diff
@@ -56,8 +56,8 @@ srecord::input_filter_message_adler32::process(const memory &input,
     // lowest address to highest.  (Holes are ignored, not filled,
     // warning issued already.)
     //
-    memory_walker_adler32::pointer w =
-        memory_walker_adler32::create();
+    auto w = std::make_shared<memory_walker_adler32>();
```

Comment:
Definitely keen to move to smart pointers. I should have picked these up when I moved from boost to std share pointers elsewhere. 

However, I'd rather move the creation of the smart pointer into the create method in the class(es). My worry is that someone using libsrecord won't get the benefit or more importantly may mix smart and non-smart pointers if they are using base and library code from the project.